protocol/morse/protocol.go

```diff
@@ -388,3 +389,36 @@ func (p *Protocol) getApps(serviceID protocol.ServiceID) ([]app, bool) {
 func sessionCacheKey(serviceID protocol.ServiceID, appAddr string) string {
 	return fmt.Sprintf("%s:%s", serviceID, appAddr)
 }
+
+// GetSanctionedEndpoints returns all Morse sanctioned endpoints
+// If serviceID is provided, it returns the sanctioned endpoints for that service ID.
+// If serviceID is not provided, it returns the sanctioned endpoints for all configured service IDs.
+// It also returns the total number of endpoints, the number of valid endpoints, and the number of sanctioned endpoints.
+func (p *Protocol) GetSanctionedEndpoints(serviceID protocol.ServiceID) devtools.SanctionDetailsResponse {
+	validEndpoints := 0
+
+	var serviceIDsToCount []protocol.ServiceID
```

Comment:
1. To what count? Please update name and/or #PUC
2. Could this be a map instead of a slice?